### PR TITLE
Extend error handling of delta issues in crawlers and hive metastore

### PR DESCRIPTION
**Pull request by mwojtyczka**
## Changes
Extend error handling of delta issues in crawlers and hive metastore by catching: `DELTA_TABLE_NOT_FOUND` and `DELTA_MISSING_TRANSACTION_LOG`

### Linked issues

Resolves [#778](https://github.com/databrickslabs/ucx/issues/778)

### Functionality 

- [x] extend error handling

### Tests
- [x] added unit tests

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/mapping.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def skip_table(self, schema: str, table: str):
                 f"ALTER TABLE `{schema}`.`{table}` SET TBLPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)"
             )
         except NotFound as nf:
-            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(nf):
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(nf) or "[DELTA_TABLE_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Table {schema}.{table}. Table not found.")
             else:
                 logger.error(nf)
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/mapping.py:108]: This error log needs context (the table name)

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py [hunk 2]
```diff
@@ -279,7 +279,7 @@ def _move_table(
 
             return True
         except NotFound as err:
-            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(err):
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(err) or "[DELTA_TABLE_NOT_FOUND]" in str(err):
                 logger.error(f"Could not find table {from_table_name}. Table not found.")
             else:
                 logger.error(err)
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/table_migrate.py:285]: This logger.error needs context (the table name)

File: src/databricks/labs/ucx/hive_metastore/table_size.py [hunk 3]
```diff
@@ -70,10 +70,11 @@ def _safe_get_table_size(self, table_full_name: str) -> int | None:
...
         except Exception as e:
-            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(e):
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(e) or "[DELTA_TABLE_NOT_FOUND]" in str(e):
                 logger.warning(f"Failed to evaluate {table_full_name} table size. Table not found.")
                 return None
             if "[DELTA_MISSING_TRANSACTION_LOG]" in str(e):
                 logger.warning(f"Delta table {table_full_name} is corrupted: missing transaction log.")
                 return None
-            raise RuntimeError(str(e)) from e
+            logger.error(e)
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/table_size.py:79]: This logger.error needs context (the table name)

**Review by dmoore247 — CHANGES_REQUESTED** [review]
The PR doesn't catch the `Py4JJavaError` that gets raised per #778 
Please ensure each log message has context (the schemaname.tablename in this case)

**Review by nfx — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/hive_metastore/table_size.py [hunk 4]
```diff
@@ -70,10 +70,11 @@ def _safe_get_table_size(self, table_full_name: str) -> int | None:
         try:
             return self._spark._jsparkSession.table(table_full_name).queryExecution().analyzed().stats().sizeInBytes()
         except Exception as e:
```
[the remark this answers is not in the document]
Comment (mwojtyczka) [reply src/databricks/labs/ucx/hive_metastore/table_size.py:72]: `table_full_name`  contains catalog, schema and table name.

Exception is the top level exception in Python. It should catch all possible errors. Without specyfing the type we also cannot print the exception.

File: src/databricks/labs/ucx/hive_metastore/mapping.py (hunk 1 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/hive_metastore/mapping.py:108]: we print schema and table name. What is missing?

File: tests/unit/framework/test_crawlers.py [hunk 5]
```diff
@@ -293,6 +307,18 @@ def test_execute(mocker):
         with pytest.raises(NotFound):
```
[the remark this answers is not in the document]
Comment (mwojtyczka) [reply tests/unit/framework/test_crawlers.py:307]: Mocking Py4JJavaError is quite complex. Shouldn't it be enough to test with Exception? (it should cover all possible errors).

File: src/databricks/labs/ucx/hive_metastore/mapping.py (hunk 1 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/hive_metastore/mapping.py]: added

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py (hunk 2 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/hive_metastore/table_migrate.py]: added

File: src/databricks/labs/ucx/hive_metastore/table_size.py (hunk 3 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/hive_metastore/table_size.py]: added

**Pull request merged**